consensus/state.go

```diff
@@ -929,7 +929,7 @@ func (cs *State) needProofBlock(height int64) bool {
 	if lastBlockMeta == nil {
 		panic(fmt.Sprintf("needProofBlock: last block meta for height %d not found", height-1))
 	}
-	return !bytes.Equal(cs.state.AppHash, lastBlockMeta.Header.AppHash)
+	return true
```

Comment:
I thought we don't need proof block anymore because all the *proof* has been in the current block already?